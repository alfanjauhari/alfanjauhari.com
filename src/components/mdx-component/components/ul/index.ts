export * from './ul';
export * from './ul.styles';
