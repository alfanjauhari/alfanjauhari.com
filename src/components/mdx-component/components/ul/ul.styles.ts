import { styled } from '@/theme';

export const StyledUl = styled('ul', {
  my: '$5',
  listStyleType: 'disc',
  pl: '$4',
  '& li': {
    my: '$2',
    pl: '$2',
    '&::marker': {
      color: '$gray4',
    },
  },
});
