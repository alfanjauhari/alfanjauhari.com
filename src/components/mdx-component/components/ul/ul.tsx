import { HTMLAttributes } from 'react';
import { StyledUl } from './ul.styles';

export type UlProps = HTMLAttributes<HTMLUListElement>;

export function Ul(props: UlProps) {
  return <StyledUl {...props} />;
}
