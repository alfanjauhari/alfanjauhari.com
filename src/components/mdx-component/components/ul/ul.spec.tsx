import { render, screen } from '@testing-library/react';
import { Ul } from './ul';

describe('MDXComponent/Ul', () => {
  it('should render successfuly', () => {
    render(
      <Ul>
        <li>List Item 1</li>
        <li>List Item 2</li>
      </Ul>,
    );

    expect(screen.getByRole('list')).toBeVisible();
    expect(screen.getAllByRole('listitem')).toHaveLength(2);
  });

  it('should match snapshot', () => {
    render(
      <Ul>
        <li>List Item 1</li>
        <li>List Item 2</li>
      </Ul>,
    );

    expect(screen.getByRole('list')).toMatchSnapshot();
  });
});
