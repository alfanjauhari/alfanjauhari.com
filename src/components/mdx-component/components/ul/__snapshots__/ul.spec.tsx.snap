// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXComponent/Ul should match snapshot 1`] = `
<ul
  class="c-ecBcCj"
>
  <li>
    List Item 1
  </li>
  <li>
    List Item 2
  </li>
</ul>
`;
